### PR TITLE
Test ability to create service without integration with CF

### PR DESCRIPTION
**Pull request by bstick12**
Create service tested

https://www.pivotaltracker.com/story/show/151720489

We have created a dummy version of the `broker.CloudFoundryClient`

### CODE REVIEW

File: integration_tests/on_demand_service_broker/binding_test.go [hunk 1]
```diff
@@ -48,19 +50,31 @@ var _ = Describe("binding service instances", func() {
...
 		boshDirector = mockbosh.NewWithUAA(boshUAA.URL)
 		cfUAA = mockuaa.NewClientCredentialsServer(cfUaaClientID, cfUaaClientSecret, "CF UAA token")
 		cfAPI = mockcfapi.New()
 
-		var brokerConfig config.Config
 		brokerConfig = defaultBrokerConfig(boshDirector.URL, boshUAA.URL, cfAPI.URL, cfUAA.URL)
 
-		runningBroker = startBrokerWithPassingStartupChecks(brokerConfig, cfAPI, boshDirector)
+		switch cfClient {
+		case "dummy":
```
Comment (nodo) [inline integration_tests/on_demand_service_broker/binding_test.go]: In reference to an above comment, would it be possible to replace "dummy" with "NoopCFClient"?

**Comment by bstick12** [comment]
@nodo All the changes should be made. Went with noopcf for package name

**Comment by nodo** [comment]
@bstick12 Please, could you rename `noopcf` to `noopservicescontroller`? Also, we can see only changes to the package name, what are your thoughts about the other comments?

File: broker/provision_test.go [hunk 2]
```diff
@@ -711,6 +712,34 @@ var _ = Describe("provisioning", func() {
...
+
+		It("succeeds", func() {
+
+			boshInfo = createBOSHInfoWithMajorVersion(
+				boshdirector.MinimumMajorSemverDirectorVersionForLifecycleErrands,
+				boshdirector.VersionType("semver"),
+			)
+			noopCFClient := noopservicescontroller.New()
+			broker, err := createBroker(boshInfo, noopCFClient)
+			Expect(err).To(BeNil())
```
Comment (nodo) [inline broker/provision_test.go]: BeNil() to HaveOccurred()

File: noopservicescontroller/client_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+			_, ok := i.(broker.CloudFoundryClient)
+			Expect(ok).To(BeTrue())
+		})
+	})
+
+	Describe("GetAPIVersion", func() {
+		It("return a valid version", func() {
+			client := noopservicescontroller.New()
+			version, err := client.GetAPIVersion(testLogger)
+			Expect(err).To(BeNil())
```
Comment (nodo) [inline noopservicescontroller/client_test.go]: BeNil() to HaveOccurred()

File: noopservicescontroller/client_test.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+			Expect(version).To(Equal(broker.MinimumCFVersion))
+		})
+	})
+
+	Describe("CountInstancesOfPlan", func() {
+		It("always returns 1", func() {
+			client := noopservicescontroller.New()
+			planCount, err := client.CountInstancesOfPlan("offeringId", "planId", testLogger)
+			Expect(err).To(BeNil())
```
Comment (nodo) [inline noopservicescontroller/client_test.go]: BeNil() to HaveOccurred()

File: noopservicescontroller/client_test.go [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+			Expect(planCount).To(Equal(1))
+		})
+	})
+
+	Describe("CountInstancesOfServiceOffering", func() {
+		It("returns empty map", func() {
+			client := noopservicescontroller.New()
+			instanceCountByPlanID, err := client.CountInstancesOfServiceOffering("offeringId", testLogger)
+			Expect(err).To(BeNil())
```
Comment (nodo) [inline noopservicescontroller/client_test.go]: BeNil() to HaveOccurred()

File: noopservicescontroller/client_test.go [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+			Expect(instanceCountByPlanID).ToNot(BeNil())
+		})
+	})
+
+	Describe("GetInstanceState", func() {
+		It("return default state", func() {
+			client := noopservicescontroller.New()
+			instanceState, err := client.GetInstanceState("serviceInstanceGUID", testLogger)
+			Expect(err).To(BeNil())
```
Comment (nodo) [inline noopservicescontroller/client_test.go]: BeNil() to HaveOccurred()

File: noopservicescontroller/client_test.go [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+			Expect(instanceState).ToNot(BeNil())
+		})
+	})
+
+	Describe("GetInstancesOfServiceOffering", func() {
+		It("gets empty instances of service offerings", func() {
+			client := noopservicescontroller.New()
+			instances, err := client.GetInstancesOfServiceOffering("serviceInstanceGUID", testLogger)
+			Expect(err).To(BeNil())
```
Comment (nodo) [inline noopservicescontroller/client_test.go]: BeNil() to HaveOccurred()

**Pull request merged**